### PR TITLE
We are now able to specify a custom port for the remote connection

### PR DESCRIPTION
**Pull request by kokeroulis**
It would be nice if we could specify the custom port in which we want to connect using the ssh.
There are some cases that the default port is not available.

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
You can do this through `~/.ssh/config`

File: mainframer.sh [hunk 1]
```diff
@@ -46,10 +51,10 @@ rsync --archive --delete --compress-level=$LOCAL_COMPRESS_LEVEL \
 --exclude='captures' \
 --exclude='**/build' \
 --exclude='**/local.properties' \
---rsh "ssh" ./ "$REMOTE_BUILD_MACHINE:~/$PROJECT_DIR_NAME"
+--rsh "ssh -p$REMOTE_SSH_PORT" ./ "$REMOTE_BUILD_MACHINE:~/$PROJECT_DIR_NAME"
```
Comment (artem-zinnatullin) [inline mainframer.sh]: Space between `-p` and `$REMOTE_SSH_PORT`?

File: mainframer.sh [hunk 2]
```diff
@@ -62,7 +67,7 @@ rsync --archive --delete --compress-level=$REMOTE_COMPRESS_LEVEL \
 --exclude='artifacts' \
 --exclude='captures' \
 --exclude='**/local.properties' \
---rsh "ssh" "$REMOTE_BUILD_MACHINE:~/$PROJECT_DIR_NAME/" ./
+--rsh "ssh -p$REMOTE_SSH_PORT" "$REMOTE_BUILD_MACHINE:~/$PROJECT_DIR_NAME/" ./
```
Comment (artem-zinnatullin) [inline mainframer.sh]: Same as above

**Review by artem-zinnatullin — CHANGES_REQUESTED** [review]

**Comment by artem-zinnatullin** [comment]
Not sure I'd like to include this since `~/.ssh/config` already allows you specify that and we actually already recommend to create separate config for remote build machine in `~/.ssh/config` to speedify connection so it's already used by most users and doesn't looks like required feature in `mainframer.sh` itself…

@ming13 thoughts?

**Comment by kokeroulis** [comment]
@artem-zinnatullin i didn't know about ~/.ssh/config, so it looks reasonable to me...

**Comment by artem-zinnatullin** [comment]
@kokeroulis general rule before spending time on creating pull request — search & create an issue :) 

i_know_that_feel_bro_by_rober_raik-d4cxn5a

**Comment by ming13** [comment]
Yep, we should delegate it to the transport itself. As a bonus, you will a get a configured SSH connection, i. e. there will be an ability to SSH to the machine without specifying the port (if I remember it correctly).

**Comment by artem-zinnatullin** [comment]
Yup. Sorry @kokeroulis, but I'm closing this!

**Comment by kokeroulis** [comment]
@artem-zinnatullin np!

**Pull request closed without merging**